src/image-copy/azext_imagecopy/custom.py

```diff
@@ -0,0 +1,135 @@
+from multiprocessing import Pool
+
+from azext_imagecopy.cli_utils import run_cli_command, prepare_cli_command
+from azext_imagecopy.create_target import create_target_image
+import azure.cli.core.azlogging as azlogging
+
+logger = azlogging.get_az_logger(__name__)
+
+
+def imagecopy(source_resource_group_name, source_object_name, target_location, \
+    target_resource_group_name, source_type='image', cleanup='false', parallel_degree=-1):
+
+    # get the os disk id from source vm/image
+    logger.warn("Getting os disk id of the source vm/image")
+    cmd = prepare_cli_command([source_type, 'show', \
+        '--name', source_object_name, \
+        '--resource-group', source_resource_group_name])
+
+    json_cmd_output = run_cli_command(cmd, True)
+
+    source_os_disk_id = json_cmd_output['storageProfile']['osDisk']['managedDisk']['id']
+    source_os_type = json_cmd_output['storageProfile']['osDisk']['osType']
+    logger.debug("source_os_disk_id: " + source_os_disk_id + " source_os_type: " + source_os_type)
+
+
+    # create source snapshots
+    logger.warn("Creating source snapshot")
+    source_os_disk_snapshot_name = source_object_name + '_os_disk_snapshot'
+    cmd = prepare_cli_command(['snapshot', 'create', \
+        '--name', source_os_disk_snapshot_name, \
+        '--resource-group', source_resource_group_name, \
+        '--source', source_os_disk_id])
+
+    run_cli_command(cmd)
+
+
+    # Get SAS URL for the snapshotName
+    logger.warn("Getting sas url for the source snapshot")
+    cmd = prepare_cli_command(['snapshot', 'grant-access', \
+        '--name', source_os_disk_snapshot_name, \
+        '--resource-group', source_resource_group_name, \
+        '--duration-in-seconds', '3600'])
+
+    json_output = run_cli_command(cmd, True)
+
+    source_os_disk_snapshot_url = json_output['accessSas']
+    logger.debug(source_os_disk_snapshot_url)
+
+
+    # Start processing in the target locations
+
+    transient_resource_group_name = 'image-copy-rg'
+    create_resource_group(transient_resource_group_name, 'eastus')
+
+    target_locations_list = target_location.split(',')
+    target_locations_count = len(target_locations_list)
+    logger.warn("Target location count: {0}".format(target_locations_count))
+
+    create_resource_group(target_resource_group_name, target_locations_list[0].strip())
+
+    if parallel_degree == -1:
+        pool = Pool(target_locations_count)
+    elif parallel_degree == 1:
+        pool = Pool(1)
+    else:
+        pool = Pool(min(parallel_degree, target_locations_count))
```

Comment:
I don't think you need `elif parallel_degree == 1:`.
min(1, target_locations_count) would be 1 anyway right since there'll be at least 1 location.